blocklylib-core/src/main/java/com/google/blockly/android/codegen/CodeGeneratorManager.java

```diff
@@ -6,34 +6,44 @@
 import android.content.ServiceConnection;
 import android.os.IBinder;
 import android.util.Log;
-import android.widget.Toast;
 
-import com.google.blockly.android.codegen.CodeGenerationRequest.CodeGeneratorCallback;
-import com.google.blockly.model.BlocklySerializerException;
-import com.google.blockly.model.Workspace;
-import com.google.blockly.utils.StringOutputStream;
-
-import java.util.List;
+import java.util.LinkedList;
+import java.util.Queue;
 
 /**
  * Client Side class responsible for connecting to the {@link CodeGeneratorService}.
+ *
+ * A connection to the service is only made the first time code generation is requested.
  */
 public class CodeGeneratorManager {
     private static final String TAG = "CodeGeneratorManager";
 
     private final Context mContext;
-    private final ServiceConnection mCodeGenerationConnection;
+    private final Queue<CodeGenerationRequest> mStoredRequests;
 
+    private final ServiceConnection mCodeGenerationConnection;
     private CodeGeneratorService mGeneratorService;
 
+    private boolean mResumed = false;
+    private boolean mIsConnecting = false;
+
     public CodeGeneratorManager(Context context) {
         this.mContext = context;
+        this.mStoredRequests = new LinkedList<>();
 
         this.mCodeGenerationConnection = new ServiceConnection() {
             @Override
             public void onServiceConnected(ComponentName className, IBinder binder) {
-                mGeneratorService =
-                    ((CodeGeneratorService.CodeGeneratorBinder) binder).getService();
+                if (!mResumed) {
+                    mContext.unbindService(mCodeGenerationConnection);
+                } else {
+                    mGeneratorService = ((CodeGeneratorService.CodeGeneratorBinder) binder).getService();
+                    mIsConnecting = false;
```

Comment:
Move `mIsConnecting = false;` to the end, outside of the `if (!mResumed)` check.  Maybe even in a `finally` block.